src/components/ToggleSlider/index.tsx

```diff
@@ -17,30 +30,54 @@ const ToggleSlider = ({ items = [], onChange }: IToggleSlider) => {
 
   const handleClick = (index: number) => {
     setActiveIndex(index);
-    onChange?.(activeIndex)
+    onChange?.(index);
   };
 
+  // Ensure layout is ready before running animations
+  useEffect(() => {
+    // Wait until the next frame to allow the DOM to fully render
+    const timeout = setTimeout(() => {
+      setReady(true);
+    }, 0);
+
+    return () => clearTimeout(timeout);
+  }, [items.length]);
+
```

Comment:
i will approve now, i think can be written better.

Reply:
Removed